File: VBNet/Node.vb
Public Class Assignment
    Public StartNode As Node
    Public FinishNode As Node
    Public AllNodes As Dictionary(Of String, Node) = New Dictionary(Of String, Node)

    Friend Sub AddEdge(first As String, distance As Integer, second As String)
        If Not AllNodes.ContainsKey(first) Then
            AllNodes.Add(first, New Node(first))
        End If
        If Not AllNodes.ContainsKey(second) Then
            AllNodes.Add(second, New Node(second))
        End If
        Dim one As Node = AllNodes(first)
        Dim two = AllNodes(second)
        one.Edges.Add(New Edge(one, two, distance))
        two.Edges.Add(New Edge(two, one, distance))
    End Sub
End Class

Public Class Node
    Public Property Name As String
    Public Property Edges As List(Of Edge) = New List(Of Edge)
    Public Property BestDistance As Integer = Integer.MaxValue
    Public Property BestParent As Node = Nothing

    Public Sub New(name As String)
        Me.Name = name
    End Sub
End Class

Public Class Edge
    Public ReadOnly Property From As Node
    Public ReadOnly Property Into As Node
    Public ReadOnly Property Distance As Integer
    Public Sub New(from As Node, into As Node, distance As Integer)
        Me.From = from
        Me.Into = into
        Me.Distance = distance
    End Sub
End Class

Public Class DijkstraResult
    Public Property Successful As Boolean
    Public Property BestPath As List(Of Node)
    Public Property PathLength As Integer
End Class

File: MultilanguageDijkstraAlgorithm/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MultilanguageDijkstraAlgorithm
{
    class Assignment
    {
        public Node StartNode { get; set; }
        public Node FinishNode { get; set; }
        public Dictionary<string, Node> AllNodes = new Dictionary<string, Node>();

        internal void AddEdge(string node1s, int distance, string node2s)
        {
            if (!AllNodes.ContainsKey(node1s))
            {
                AllNodes.Add(node1s, new Node(node1s));
            }
            if (!AllNodes.ContainsKey(node2s))
            {
                AllNodes.Add(node2s, new Node(node2s));
            }
            Node one = AllNodes[node1s];
            Node two = AllNodes[node2s];
            one.Edges.Add(new Edge(one, two, distance));
            two.Edges.Add(new Edge(two, one, distance));
        }
    }
    class Node
    {
        public string Name { get; set; }
        public List<Edge> Edges = new List<Edge>();
        public int BestDistance = Int32.MaxValue;
        public Node BestParent = null;

        public Node(string name)
        {
            this.Name = name;
        }
    }
    class Edge
    {
        public Node From { get; set; }
        public Node To { get; set; }
        public int Distance { get; set; }
        public Edge(Node from, Node to, int distance)
        {
            From = from;
            To = to;
            Distance = distance;
        }
    }
    class DijkstraResult
    {
        public bool Successful { get; set; }
        public List<Node> BestPath = new List<Node>();
        public int PathLength;
    }
}
